In [10]:
from czifile import CziFile
import numpy as np

In [2]:
pwd

'/Users/morganschwartz/Code/germband-extension/notebooks'

In [3]:
datadir = '../data/20180108_htl_glc_sc11_mmzm.czi'

The syntax that follows below was taken from this  [tutorial] posted on *Scientific Programming with Python*.

[tutorial]: (http://schryer.github.io/python_course_material/python/python_10.html)

In [4]:
with CziFile(datadir) as czi:
    raw_im = czi.asarray()

In [5]:
type(raw_im),raw_im.shape

(numpy.ndarray, (1, 1, 2, 156, 1, 1024, 1024, 1))

The output object is a `numpy` array with 8 dimensions. Based on some notes in the [tutorial], we can guess what each dimension most likely contains.

`(?, roi, channel, time, z, x, y, ?)`

For this particular data, we collected a timelapse movie with 2 channels (GFP, brightfield). For this analysis, we only need the brightfield channel so we will index to select just the data that we want.

In [9]:
data = raw_im[:,:,1,:,:,:,:,:]
data.shape

(1, 1, 156, 1, 1024, 1024, 1)

Our new array contains 156 timepoints, 1 channel, and xy dimensions of 1024x1024. We would like to reduce this array to the minimum number of dimensions (3), which we can do with the `np.squeeze` function.

In [12]:
data = np.squeeze(data)
data.shape

(156, 1024, 1024)

Now we have the minimal number of dimensions in our array and we are ready to save the data in whatever format we prefer.

# Test utility function in gbeflow

In [14]:
from imp import reload
import gbeflow

In [52]:
reload(gbeflow)

<module 'gbeflow' from '/Users/morganschwartz/Code/germband-extension/gbeflow/__init__.py'>

The `gbeflow.CziImport` function reads in the file specified in the input and saves the squeezed array in the attribute `data`.

In [53]:
czi = gbeflow.CziImport(datadir)


            There are 1 ROIs,
                       2 channels and
                       156 timepoints.
            

            The 3D dimensions of the data are:
            1 x 1024 x 1024 (zyx)
            


In [55]:
czi.data.shape

(2, 156, 1024, 1024)

In [6]:
%load_ext watermark

In [7]:
%watermark -v -p czifile

CPython 3.7.0
IPython 6.5.0

czifile 2018.10.18
